client/state/data-layer/wpcom/comments/index.js

```diff
@@ -36,6 +63,38 @@ export const fetchPostComments = ( { dispatch, getState }, action ) => {
 	}, action ) );
 };
 
+export const writePostComment = ( { dispatch }, action ) => {
+	const { siteId, postId, parentCommentId, commentText } = action;
+	const placeholder = createPlaceholderComment( commentText, postId, parentCommentId );
+
+	// Insert a placeholder
+	dispatch( {
+		type: COMMENTS_RECEIVE,
+		siteId,
+		postId,
+		comments: [ placeholder ],
+		skipSort: !! parentCommentId
+	} );
+
+	const path = !! parentCommentId
+		? `/sites/${ siteId }/comments/${ parentCommentId }/replies/new`
+		: `/sites/${ siteId }/posts/${ postId }/replies/new`;
```

Comment:
well honestly this is an exception because our API follows some awkward conventions. as a "REST API" it would map out fairly well. again, we're not trying to create new APIs here but clearly define the interface between the WordPress.com API and Calypso.

I'm struggling a bit too with `resources` but I'm willing to "let the complexity bleed" here because it's closer to the reality of the API we're hitting.

Otherwise this isn't complicated: duplicate the code in both places but filter on which is which. I almost suggested this before. This would leave us with `wpcom/sites/posts/replies/new/index.js` and `wpcom/sites/comments/replies/new/index.js` and probably a shared utility dependency they import.

Maybe with all this discussion and confusion it would simply be best to duplicate for now, as this seems to be the exception rather than the norm.